sdk/core/Azure.Core/src/Diagnostics/AzureCoreEventSource.cs

```diff
@@ -154,5 +156,13 @@ public void RequestRedirectCountExceeded(string requestId, string from, string t
         {
             WriteEvent(RequestRedirectCountExceededEvent, requestId, from, to);
         }
+
+        [Event(
+            PipelineTransportOptionsNotAppliedEvent,
+            Level = EventLevel.Warning, Message = "PipelineTransportOptions where provided for options type [{0}] but were not applied because a custom Transport has already been set.")]
```

Comment:
Wonder if we need to make this message a little bit more user-facing. They don't know what `PipelineTransportOptions` is and why it matters.

Something like `The client requires transport configuration but it was not applied because custom transport was provided`.

`(Type optionsType)`

I think method types need to match event types.

Let's start with this being Info and see if we start getting reports about customers broken by it. There is no way to suppress this warning event if the customer provided transport that satisfies client requirements.